db/db_versioned.go

```diff
@@ -196,6 +199,241 @@ func (b *BoltDBVersioned) Version(ns string, key []byte) (uint64, error) {
 	return last, err
 }
 
+// CommitToDB write a batch to DB, where the batch can contain keys for
+// both versioned and non-versioned namespace
+func (b *BoltDBVersioned) CommitToDB(version uint64, vns map[string]bool, kvsb batch.KVStoreBatch) error {
+	vnsize, ve, nve, err := dedup(vns, kvsb)
```

Comment:
split in caller side

Reply:
that's doable, then the interface would become 
```
(version uint64, vnsize map[string]int, ve, nve []*batch.WriteInfo)
```
feels a bit clumsy and too low-level, the current interface is cleaner IMO